xray/default_emitter.go

```
@@ -0,0 +1,103 @@
+// Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the License. A copy of the License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+package xray
+
+import (
+	"bytes"
+	"encoding/json"
+	"net"
+	"sync"
+
+	log "github.com/cihub/seelog"
+)
+
+// Header is added before sending segments to daemon.
+var Header = []byte(`{"format": "json", "version": 1}` + "\n")
+
+// DefaultEmitter provides the naive implementation of emitting trace entities.
+type DefaultEmitter struct {
+	sync.Mutex
+	conn *net.UDPConn
+}
+
+// NewDefaultEmitter initializes and returns a
+// pointer to an instance of DefaultEmitter.
+func NewDefaultEmitter(raddr *net.UDPAddr) (*DefaultEmitter, error) {
+	initLambda()
+	d := &DefaultEmitter{}
+	d.RefreshEmitterWithAddress(raddr)
+	return d, nil
+}
+
+// RefreshEmitterWithAddress dials UDP based on the input UPD address.
+func (de *DefaultEmitter) RefreshEmitterWithAddress(raddr *net.UDPAddr) {
+	de.Lock()
+	de.conn, _ = net.DialUDP("udp", nil, raddr)
```

Could we don't ignore the error when calling the DialUDP method?